doc/doc_environment.yml

```diff
@@ -0,0 +1,23 @@
+name: sagemaker-docs
```

Comment:
how does this new file interact with `requirements.txt`? do we have to keep dependencies up-to-date with both?

Reply:
It doesn't; as far as i could tell, requirements.txt is for tox. I created that yaml approach because it's a really nice, repeatable way for anyone to setup their docs environment. How that can dovetail with tox, I don't know yet. Ideally, we don't have deps sprinkled about in different places.

Reply:
how about for the scope of this PR we remove this file for now, and then in another one we can tackle keeping all the dependencies in one place?

Reply:
Sure, i'll make that a separate PR.